src/lib.rs

```diff
@@ -1198,36 +1174,68 @@ decl_module! {
         }
 
         /// submit a poll
-        // fn submit_poll(origin, thread_id: T::ThreadId, poll_value: PollData) -> dispatch::Result {
-        //     // Check that its a valid signature
-        //     let who = ensure_signed(origin)?;
-
-        //     // get forum user id.
-        //     let forum_user_id = Self::ensure_is_forum_member_with_correct_account(&who)?;
-
-        //     // Get thread
-        //     let thread = Self::ensure_thread_exists(&thread_id)?;
-
-        //     // Make sure poll exist and not expired
-        //     match thread.poll {
-        //         None => Err(ERROR_POLL_NOT_EXIST),
-        //         Some(poll) => {
-        //             if poll.end_time < <timestamp::Module<T>>::now() {
-        //                 Err(ERROR_POLL_COMMIT_EXPIRED)
-        //             } else {
-        //                 Self::ensure_poll_data_valid(&poll, poll_value)?;
-        //                 <PollDataByAccount<T>>::insert(thread_id, forum_user_id, poll_value);
-        //                 if <PollStatistics<T>>::exists(thread_id, poll_value) {
-        //                     <PollStatistics<T>>::insert(thread_id, poll_value, 1);
-        //                 } else {
-        //                     let old_value = <PollStatistics<T>>::get(thread_id, poll_value);
-        //                     <PollStatistics<T>>::insert(thread_id, poll_value, old_value + 1);
-        //                 }
-        //                 Ok(())
-        //             }
-        //         }
-        //     }
-        // }
+        fn submit_poll(origin, forum_user_id: T::ForumUserId, thread_id: T::ThreadId, index: u32) -> dispatch::Result {
```

Comment:
Can you change name to be clearer, such as `vote_on_poll`.

Reply:
accepted.